debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresChangeRecordEmitter.java

```diff
@@ -188,12 +195,65 @@ private Object[] columnValues(List<ReplicationMessage.Column> columns, TableId t
                         }
                     }
                 }
-                values[position] = value;
+
+                if (connectorConfig.plugin().isYBOutput()) {
+                    // YB Note: In this case, if we have the plugin yboutput and the column contains
+                    // the unchanged toasted value, we will not form a value struct for it.
+                    // Ultimately, it will be emitted as a NULL value.
+                    if (!UnchangedToastedReplicationMessageColumn.isUnchangedToastedValue(value)) {
+                        values[position] = new Object[]{value, Boolean.TRUE};
+                    }
+                } else {
+                    LOGGER.debug("Plugin is NOT yboutput");
+                    values[position] = value;
+                }
             }
         }
         return values;
     }
 
+    @Override
+    protected void emitUpdateRecord(Receiver<PostgresPartition> receiver, TableSchema tableSchema) throws InterruptedException {
+        Object[] oldColumnValues = getOldColumnValues();
+        Object[] newColumnValues = getNewColumnValues();
+
+        Struct oldKey = tableSchema.keyFromColumnData(oldColumnValues);
+        Struct newKey = tableSchema.keyFromColumnData(newColumnValues);
+
+        Struct newValue = tableSchema.valueFromColumnData(newColumnValues);
+        Struct oldValue = tableSchema.valueFromColumnData(oldColumnValues);
+
+        if (skipEmptyMessages() && (newColumnValues == null || newColumnValues.length == 0)) {
+            LOGGER.debug("no new values found for table '{}' from update message at '{}'; skipping record", tableSchema, getOffset().getSourceInfo());
+            return;
+        }
+
+        /*
+         * If skip.messages.without.change is configured true,
+         * Skip Publishing the message in case there is no change in monitored columns
+         * (Postgres) Only works if REPLICA IDENTITY is set to FULL - as oldValues won't be available
+         */
+        if (skipMessagesWithoutChange() && Objects.nonNull(newValue) && newValue.equals(oldValue)) {
```

Comment:
It would be good to test it once for crash with CHANGE

Reply:
Added two tests:
* `shouldNotSkipMessagesWithoutChangeWithReplicaIdentityChange`
* `shouldSkipMessagesWithoutChangeWithReplicaIdentityFull`

The situation is that we have only provided 2 of the 3 columns in the `column.include.list` and now if the non-monitored column is updated, the behaviour would be as follows:
* `CHANGE`: Since `oldValue` would be null, the condition `newValue.equals(oldValue)` will because false and we will not skip the message.
* `FULL`: Since `oldValue` would have the same content as `newValue` as the column which is actually changed is not monitored, the condition `newValue.equals(oldValue)` will become true and we will skip this message.